src/coreclr/jit/fgprofile.cpp

```diff
@@ -407,11 +419,11 @@ void BlockCountInstrumentor::InstrumentMethodEntry(Schema& schema, BYTE* profile
 
     // Find the address of the entry block's counter.
     //
-    BasicBlock* const block            = m_comp->fgFirstBB;
-    const int         firstSchemaIndex = block->bbCountSchemaIndex;
-    assert(block->bbCodeOffs == (IL_OFFSET)schema[firstSchemaIndex].ILOffset);
+    const int firstSchemaIndex = (int)m_entryBlock->bbCountSchemaIndex;
+    assert(m_entryBlock->bbCodeOffs == (IL_OFFSET)schema[firstSchemaIndex].ILOffset);
```

Comment:
Wouldn't this be clearer if we asserted that both are zero?

```
assert(m_entryBlock->bbCodeOffs == 0);
assert(schema[firstSchemaIndex].ILOffset == 0);
```

Reply:
Also do we need to assert that `m_entryBlock` is not nullptr?

Reply:
Thanks -- made updates.